static/js/stats/stats.js

```diff
@@ -282,12 +290,42 @@ function populate_messages_sent_over_time(data) {
     }
 }
 
+function format_date_for_last_update(last_update) {
+    last_update = new Date(last_update * 1000);
+
+    var hour = last_update.getHours();
+    var minutes = last_update.getMinutes();
+
+    var hour_str;
+    if (hour === 0) {
+        hour_str = '12' + ':' + minutes + 'AM';
+    } else if (hour === 12) {
+        hour_str = '12' + ':' + minutes + 'PM';
+    } else if (hour < 12) {
+        hour_str = hour + ':' + minutes + ' AM';
+    } else {
+        hour_str = (hour-12) + ':' + minutes + ' PM';
+    }
+
+    last_update = hour_str + ' on ' + format_date(last_update, false);
+    return last_update;
+}
+
 $.get({
     url: '/json/analytics/chart_data',
     data: {chart_name: 'messages_sent_over_time', min_length: '10'},
     idempotent: true,
     success: function (data) {
         populate_messages_sent_over_time(data);
+
+        // Last Full Update
```

Comment:
This is not guaranteed to return last of the 4 charts. So we'll need to find some way of updating the `.last_full_update` text regardless of the order the `$.get` calls return. 

I'd suggest just updating the text each time a new graph comes in, i.e. putting the last 4 lines of this hunk in `update_last_full_update`.